src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -406,23 +407,32 @@ private IEnumerable<JsonPropertySerialization> GetPropertySerializationsFromBag(
         public JsonObjectSerialization BuildJsonObjectSerialization(InputModelType inputModel, SchemaObjectType objectType)
         {
             var propertyBag = new SerializationPropertyBag();
+            var selfPropertyBag = new SerializationPropertyBag();
             foreach (var objectTypeLevel in objectType.EnumerateHierarchy())
             {
                 foreach (var objectTypeProperty in objectTypeLevel.Properties)
                 {
                     if (objectTypeProperty == objectTypeLevel.AdditionalPropertiesProperty)
                         continue;
                     propertyBag.Properties.Add(objectTypeProperty, objectType.GetForMemberSerialization(objectTypeProperty.Declaration.Name));
+
+                    if (objectTypeLevel == objectType)
+                    {
+                        selfPropertyBag.Properties.Add(objectTypeProperty, objectType.GetForMemberSerialization(objectTypeProperty.Declaration.Name));
+                    }
                 }
             }
             PopulatePropertyBag(propertyBag, 0);
+            PopulatePropertyBag(selfPropertyBag, 0);
+
             var properties = GetPropertySerializationsFromBag(propertyBag, objectType).ToArray();
+            var selfProperties = GetPropertySerializationsFromBag(selfPropertyBag, objectType).ToArray();
```

Comment:
Please add some comments here to explain why we have to have two "property bags" here. This looks so weird